packages/react-next/src/components/Link/useLink.ts

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { ComposePreparedOptions } from '@fluentui/react-compose';
+import { classNamesFunction } from '../../Utilities';
+import { ILink, ILinkProps, ILinkStyleProps, ILinkStyles } from './Link.types';
+
+const getClassNames = classNamesFunction<ILinkStyleProps, ILinkStyles>({ useStaticStyles: true });
+
+/**
+ * The useLink hook processes the Link component props and returns
+ * state, slots and slotProps for consumption by the component.
+ */
+export const useLink = (props: ILinkProps, options: ComposePreparedOptions): any => {
+  const { as, className, disabled, href, onClick, ref, styles, theme } = props;
+
+  useComponentRef(props, ref);
+
+  const classNames = getClassNames(styles!, {
```

Comment:
it seems weird for this to be in `useLink`.

Reply:
I could move it out into the render function itself, but basically it is needed because we are still using `styled`. The drawback as we said before is that we still have styles on the base component.

Reply:
`Toggle` also has this logic inside `useToggle` so if we're moving this outside of `useLink` we should probably move that as well.

Reply:
easy to move later!